Mathlib/Analysis/CStarAlgebra/ApproximateUnit.lean

```diff
@@ -105,3 +112,195 @@ lemma CStarAlgebra.directedOn_nonneg_ball :
       have hab' : cfcₙ g a ≤ cfcₙ g a + cfcₙ g b := le_add_of_nonneg_right cfcₙ_nonneg_of_predicate
       exact CFC.monotoneOn_one_sub_one_add_inv cfcₙ_nonneg_of_predicate
         (cfcₙ_nonneg_of_predicate.trans hab') hab'
+
+section ApproximateUnit
+
+open Metric Filter Topology
+
+/-- An *increasing approximate unit* in a C⋆-algebra is an approximate unit contained in the
+closed unit ball of nonnegative elements. -/
+structure Filter.IsIncreasingApproximateUnit (l : Filter A) extends l.IsApproximateUnit : Prop where
+  eventually_nonneg : ∀ᶠ x in l, 0 ≤ x
+  eventually_norm : ∀ᶠ x in l, ‖x‖ ≤ 1
+
```

Comment:
Should we have a notion of "increasing filter" somewhere that says `∀ᶠ x in l, ∀ᶠ y in l, x ≤ y`? Having the notion defined in a round-about way like this makes me a bit uneasy. Presumably we'll need this property anyway, no?

Reply:
By "round-about way", do you mean that we're getting it because the nonnegative elements in the unit ball in a C⋆-algebra form a directed set?

If so, I understand, but I don't know that adding this increasing condition is better. It seems that maybe it's just a problem with the language? As for naming, I took this from Bourbaki:

> Soit `A` une algèbre normée. Une *unité approchée* de `A` est une base de filtre `𝓕` sue la boule unité de `A` telle que, pour tout `x` dans `A`, les bases de filtre `x𝓕` et `𝓕x` sur `A` convergent vers `x`.
> ...
> Si `A` est une algèbre stellaire, une unité approchée `𝓕` est dite **croissante** si `𝓕` est une base de filtre sur `{x : A | 0 ≤ A}`.
>
> Bourbaki (2019), Théories spectrales I §6 No. 10 Definition 7

Reply:
By "round-about way", I mean that it's not immediately obvious why this  should even be called "increasing". To me the idea that a filter is increasing should intuitively mean what I wrote above, but I'm not even sure it's equivalent to this definition in this case. I guess I do have the same issue with the Bourbaki definition.

Reply:
It's called "increasing" because these filters arise from monotone nets. Given any monotone net of nonnegative elements which is cofinal in the directed set of nonnegative elements in the unit ball, the filter of sections corresponding to this net (restricted to the closed unit ball) is an increasing approximate unit in the sense of this PR.

Reply:
Right -- so I guess the definition should be equivalent to `∀ᶠ x in l, ∀ᶠ y in l, x ≤ y`, no?

Reply:
No? That holds for the filter of sections for any increasing sequence (they wouldn't even need to be positive or contained in the unit ball). 

Your statement is *valid* for increasing approximate units, but it doesn't *characterize* them.

Reply:
Yes, of course that statement alone wouldn't give you positivity or the norm bound, but it does seem like that's what "increasing" should mean. Anyway, I guess that's just a terminology quibble, and I suppose we should just go with the literature anyway.